specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2018-02-14/keyvault.json

```diff
@@ -665,6 +665,187 @@
           "application/json"
         ]
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults/{vaultName}/privateEndpointConnections/{privateEndpointConnectionName}": {
```

Comment:
>/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults/{vaultName}/privateEndpointConnections [](start = 5, length = 141)

There doesn't appear to be a LIST operation for `privateEndpointConnections`. How would a user discover what they to a GET on in the first place?

Reply:
This was copied from https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json. They also don't have LIST.

Why the storage code was accepted? Is this a new requirement?

Reply:
This should have a LIST operation. Nested types require both Get By Name and LIST. The fact that this was missed in another PR doesn't change that. We can work with SRP to fix.

Reply:
@KrisBash @majastrz 
Per Azure Network team definition, the LIST operation is omitted because RPs (including Storage) must include the list of `privateEndpointConnections` in the properties of the master resource. SRP does that in the definition of `StorageAccountProperties`: https://github.com/Azure/azure-rest-api-specs/blob/68831c6eded64f98fabe9eb658712ceff16c6a2d/specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json#L1996

This PR is doing the same; see the definition of `VaultProperties`.

AKV is not adding LIST now as it does not seem necessary and would never be used by any client. If you have concerns on the design of always including child items in the properties of master resource, please open a thread with SRP, AKV and Azure Network, but don't block this PR.

Reply:
<while I am not following full thread here>
The feedback is per Azure guidelines and for better customer experience (to not let them force different ways of doing things for different RP / types). I am sure we all agree that is important, instead of following what SRP did in past.

---
In reply to: [363992368](https://github.com/Azure/azure-rest-api-specs/pull/7631#discussion_r363992368) [](ancestors = 363992368)

Reply:
I hear you. As explained, I don't own API design of Private Links. I just own the product that needs to integrate. Some of these APIs are be called by NRP, so we have little room to change.

IMHO, the problems appeared when changes to SRP were approved in Swagger. They are now in `stable` state, NRP is deployed globally, and I don't think this would ever change in near future.